src/frontend/src/lib/utils/transaction.utils.ts

```diff
@@ -29,3 +34,34 @@ export const mapTransactionIcon = ({
 					? IconSend
 					: IconReceive;
 };
+
+/**
+ * Group a list of transactions by date.
+ *
+ * It assumes that the transactions are already sorted by timestamp in descending order.
+ * It does not sort the transactions, nor the groups.
+ * If a transaction has no timestamp, it will be grouped under the key 'undefined'.
+ *
+ * @param transactions - List of transactions to group.
+ * @returns Object where the keys are the date and the values are the transactions for that date.
+ */
+export const groupTransactionsByDate = <
+	T extends BtcTransactionUi | EthTransactionUi | IcTransactionUi
+>(
+	transactions: T[]
+) => {
+	const currentDate = new Date();
+
+	return transactions.reduce<Record<string, T[]>>((acc, transaction) => {
+		if (isNullish(transaction.timestamp)) {
+			return { ...acc, undefined: [...(acc['undefined'] ?? []), transaction] };
+		}
+
+		const date = formatSecondsToNormalizedDate({
+			seconds: Number(transaction.timestamp),
```

Comment:
We are sure the timestamp is always seconds here?

Reply:
So, I would have managed this when using this function: the list of transaction provided would be all with timestamp in seconds; but I can put a sort of guard here: `if timestamp is same order of now*1000 or now*1000000 ....`

Reply:
I would not modify the transactions to their core to make them all in seconds (BTW, only IC transactions are in nanoseconds)

Reply:
If we are sure those are seconds it's fine, I was just thinking that it would be nice to not start displaying date like Dec 20 3098 or Jan 1 1970. So if there is a risk, yeah maybe a guard would be nice.